### ISSUE TITLE
Random errors when accessing the metrics ui

### ISSUE THREAD

**Issue by davidfowl** [dotnet/aspire#5095]
Got 8.1.0 , made a new aspire starter with redis and ran it. Navigated around a bit and then saw this.

Console output

```
fail: Aspire.Hosting.Dcp.dcpctrl.IdeExecutableRunner[0]
      failed to read message from IDE run session notification endpoint, recycling connection...        {"error": "read tcp [::1]:59880->[::1]:54462: i/o timeout"}
fail: Aspire.Hosting.Dcp.dcpctrl.IdeExecutableRunner[0]
      failed to read message from IDE run session notification endpoint, recycling connection...        {"error": "read tcp [::1]:61484->[::1]:54462: i/o timeout"}
fail: Aspire.Hosting.Dashboard.Microsoft.AspNetCore.Components.Server.Circuits.CircuitHost[111]
      Unhandled exception in circuit 'lojKaj4eL9BbE4uHgdi_zMC64bYevrfE1Y_xHqEA8uY'.
      Microsoft.JSInterop.JSException: Cannot read properties of null (reading 'replaceChildren')
TypeError: Cannot read properties of null (reading 'replaceChildren')
    at Module.initializeChart (https://localhost:17016/js/app-metrics.js:11:23)
    at https://localhost:17016/_framework/blazor.web.js:1:3244
    at new Promise (<anonymous>)
    at y.beginInvokeJSFromDotNet (https://localhost:17016/_framework/blazor.web.js:1:3201)
    at gn._invokeClientMethod (https://localhost:17016/_framework/blazor.web.js:1:62841)
    at gn._processIncomingData (https://localhost:17016/_framework/blazor.web.js:1:60316)
    at connection.onreceive (https://localhost:17016/_framework/blazor.web.js:1:53957)
    at i.onmessage (https://localhost:17016/_framework/blazor.web.js:1:82102)
         at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
         at Microsoft.JSInterop.JSObjectReferenceExtensions.InvokeVoidAsync(IJSObjectReference jsObjectReference, String identifier, Object[] args)
         at Aspire.Dashboard.Components.PlotlyChart.OnChartUpdated(List`1 traces, List`1 xValues, List`1 exemplars, Boolean tickUpdate, DateTimeOffset inProgressDataTime) in /_/src/Aspire.Dashboard/Components/Controls/Chart/PlotlyChart.razor.cs:line 78
         at Aspire.Dashboard.Components.Controls.Chart.ChartBase.UpdateChart(Boolean tickUpdate, DateTimeOffset inProgressDataTime) in /_/src/Aspire.Dashboard/Components/Controls/Chart/ChartBase.cs:line 505
         at Aspire.Dashboard.Components.Controls.Chart.ChartBase.OnAfterRenderAsync(Boolean firstRender) in /_/src/Aspire.Dashboard/Components/Controls/Chart/ChartBase.cs:line 93
         at Aspire.Dashboard.Components.PlotlyChart.OnAfterRenderAsync(Boolean firstRender) in /_/src/Aspire.Dashboard/Components/Controls/Chart/PlotlyChart.razor.cs:line 168
         at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask(Task taskToHandle, ComponentState owningComponentState)
```

Browser console log

```
 [2024-07-27T03:35:39.652Z] Information: Normalizing '_blazor' to 'https://localhost:17016/_blazor'.
 [2024-07-27T03:35:39.723Z] Information: WebSocket connected to wss://localhost:17016/_blazor?id=ES3oDSfYuSnQn1CnO4kX9Q.
 Attribute direction has changed from column to column.
 Attribute direction has changed from null to column.
 Attribute collapsed has changed from  to .
 Attribute collapsed has changed from null to .
 Attribute barsize has changed from null to 5.
 Attribute slot1minsize has changed from null to 150px.
 Attribute slot2minsize has changed from null to 150px.
 Attribute direction has changed from column to column.
 Attribute direction has changed from null to column.
 Attribute collapsed has changed from  to .
 Attribute collapsed has changed from null to .
 Attribute barsize has changed from null to 5.
 Attribute slot1minsize has changed from null to 150px.
 Attribute slot2minsize has changed from null to 150px.
 Attribute collapsed has changed from  to null.
:17016/traces/detail/70b8f0cfc51a629518e922695cc07a65:1 Autofocus processing was blocked because a document already has a focused element.
 Attribute collapsed has changed from  to .
 Attribute collapsed has changed from null to .
 Attribute direction has changed from column to column.
 Attribute direction has changed from null to column.
 Attribute collapsed has changed from  to .
 Attribute collapsed has changed from null to .
 Attribute barsize has changed from null to 5.
 Attribute slot1minsize has changed from null to 150px.
 Attribute slot2minsize has changed from null to 150px.
 Attribute direction has changed from column to column.
 Attribute direction has changed from null to column.
 Attribute collapsed has changed from  to .
 Attribute collapsed has changed from null to .
 Attribute barsize has changed from null to 5.
 Attribute slot1minsize has changed from null to 150px.
 Attribute slot2minsize has changed from null to 150px.
 Attribute direction has changed from column to column.
 Attribute direction has changed from null to column.
 Attribute collapsed has changed from  to .
 Attribute collapsed has changed from null to .
 Attribute barsize has changed from null to 5.
 Attribute slot1minsize has changed from null to 150px.
 Attribute slot2minsize has changed from null to 150px.
 Attribute direction has changed from column to column.
 Attribute direction has changed from null to column.
 Attribute collapsed has changed from  to .
 Attribute collapsed has changed from null to .
 Attribute barsize has changed from null to 5.
 Attribute slot1minsize has changed from null to 150px.
 Attribute slot2minsize has changed from null to 150px.
 Attribute collapsed has changed from  to null.
 Attribute direction has changed from row to row.
 Attribute direction has changed from column to row.
 Attribute direction has changed from column to column.
 Attribute direction has changed from row to column.
 Attribute collapsed has changed from  to .
 Attribute collapsed has changed from null to .
 Attribute direction has changed from column to column.
 Attribute direction has changed from null to column.
 Attribute collapsed has changed from  to .
 Attribute collapsed has changed from null to .
 Attribute barsize has changed from null to 5.
 Attribute slot1minsize has changed from null to 150px.
 Attribute slot2minsize has changed from null to 150px.
  [2024-07-27T03:51:47.172Z] Error: Connection disconnected with error 'Error: WebSocket closed with status code: 1006 (no reason given).'.
log @ blazor.web.js:1
 [2024-07-27T04:40:45.974Z] Information: Normalizing '_blazor' to 'https://localhost:17016/_blazor'.
 [2024-07-27T04:40:46.163Z] Information: WebSocket connected to wss://localhost:17016/_blazor?id=czIV_GofRktgyOUKZSXxIA.
  [2024-07-27T04:40:46.497Z] Error: Connection disconnected with error 'Error: WebSocket closed with status code: 1006 (no reason given).'.
log @ blazor.web.js:1
 [2024-07-27T04:41:05.564Z] Information: Normalizing '_blazor' to 'https://localhost:17016/_blazor'.
 [2024-07-27T04:41:07.992Z] Information: WebSocket connected to wss://localhost:17016/_blazor?id=L-whb0yG1KgW70fW3vMp9Q.
SplitPanels.ts:193 Attribute direction has changed from column to column.
SplitPanels.ts:193 Attribute direction has changed from null to column.
SplitPanels.ts:193 Attribute collapsed has changed from  to .
SplitPanels.ts:193 Attribute collapsed has changed from null to .
SplitPanels.ts:193 Attribute barsize has changed from null to 5.
SplitPanels.ts:193 Attribute slot1minsize has changed from null to 150px.
SplitPanels.ts:193 Attribute slot2minsize has changed from null to 150px.
SplitPanels.ts:193 Attribute direction has changed from column to column.
SplitPanels.ts:193 Attribute direction has changed from null to column.
SplitPanels.ts:193 Attribute collapsed has changed from  to .
SplitPanels.ts:193 Attribute collapsed has changed from null to .
SplitPanels.ts:193 Attribute barsize has changed from null to 5.
SplitPanels.ts:193 Attribute slot1minsize has changed from null to 150px.
SplitPanels.ts:193 Attribute slot2minsize has changed from null to 150px.
SplitPanels.ts:193 Attribute direction has changed from column to column.
SplitPanels.ts:193 Attribute direction has changed from null to column.
SplitPanels.ts:193 Attribute collapsed has changed from  to .
SplitPanels.ts:193 Attribute collapsed has changed from null to .
SplitPanels.ts:193 Attribute barsize has changed from null to 5.
SplitPanels.ts:193 Attribute slot1minsize has changed from null to 150px.
SplitPanels.ts:193 Attribute slot2minsize has changed from null to 150px.
SplitPanels.ts:193 Attribute collapsed has changed from  to null.
SplitPanels.ts:193 Attribute collapsed has changed from  to .
SplitPanels.ts:193 Attribute collapsed has changed from null to .
SplitPanels.ts:193 Attribute direction has changed from column to column.
SplitPanels.ts:193 Attribute direction has changed from null to column.
SplitPanels.ts:193 Attribute collapsed has changed from  to .
SplitPanels.ts:193 Attribute collapsed has changed from null to .
SplitPanels.ts:193 Attribute barsize has changed from null to 5.
SplitPanels.ts:193 Attribute slot1minsize has changed from null to 150px.
SplitPanels.ts:193 Attribute slot2minsize has changed from null to 150px.
SplitPanels.ts:193 Attribute direction has changed from row to row.
SplitPanels.ts:193 Attribute direction has changed from null to row.
SplitPanels.ts:193 Attribute barsize has changed from null to 5.
SplitPanels.ts:193 Attribute direction has changed from column to column.
SplitPanels.ts:193 Attribute direction has changed from null to column.
SplitPanels.ts:193 Attribute collapsed has changed from  to .
SplitPanels.ts:193 Attribute collapsed has changed from null to .
SplitPanels.ts:193 Attribute barsize has changed from null to 5.
SplitPanels.ts:193 Attribute slot1minsize has changed from null to 150px.
SplitPanels.ts:193 Attribute slot2minsize has changed from null to 150px.
blazor.web.js:1  [2024-07-27T04:42:43.624Z] Error: Microsoft.JSInterop.JSException: Cannot read properties of null (reading 'replaceChildren')
TypeError: Cannot read properties of null (reading 'replaceChildren')
    at Module.initializeChart (https://localhost:17016/js/app-metrics.js:11:23)
    at https://localhost:17016/_framework/blazor.web.js:1:3244
    at new Promise (<anonymous>)
    at y.beginInvokeJSFromDotNet (https://localhost:17016/_framework/blazor.web.js:1:3201)
    at gn._invokeClientMethod (https://localhost:17016/_framework/blazor.web.js:1:62841)
    at gn._processIncomingData (https://localhost:17016/_framework/blazor.web.js:1:60316)
    at connection.onreceive (https://localhost:17016/_framework/blazor.web.js:1:53957)
    at i.onmessage (https://localhost:17016/_framework/blazor.web.js:1:82102)
   at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
   at Microsoft.JSInterop.JSObjectReferenceExtensions.InvokeVoidAsync(IJSObjectReference jsObjectReference, String identifier, Object[] args)
   at Aspire.Dashboard.Components.PlotlyChart.OnChartUpdated(List`1 traces, List`1 xValues, List`1 exemplars, Boolean tickUpdate, DateTimeOffset inProgressDataTime) in /_/src/Aspire.Dashboard/Components/Controls/Chart/PlotlyChart.razor.cs:line 78
   at Aspire.Dashboard.Components.Controls.Chart.ChartBase.UpdateChart(Boolean tickUpdate, DateTimeOffset inProgressDataTime) in /_/src/Aspire.Dashboard/Components/Controls/Chart/ChartBase.cs:line 505
   at Aspire.Dashboard.Components.Controls.Chart.ChartBase.OnAfterRenderAsync(Boolean firstRender) in /_/src/Aspire.Dashboard/Components/Controls/Chart/ChartBase.cs:line 93
   at Aspire.Dashboard.Components.PlotlyChart.OnAfterRenderAsync(Boolean firstRender) in /_/src/Aspire.Dashboard/Components/Controls/Chart/PlotlyChart.razor.cs:line 168
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask(Task taskToHandle, ComponentState owningComponentState)
log @ blazor.web.js:1
blazor.web.js:1 [2024-07-27T04:42:43.625Z] Information: Connection disconnected.
blazor.web.js:1  Uncaught (in promise) Error: Cannot send data if the connection is not in the 'Connected' State.
    at Nn.send (blazor.web.js:1:85376)
    at gn._sendMessage (blazor.web.js:1:58807)
    at gn._sendWithProtocol (blazor.web.js:1:58897)
    at gn.send (blazor.web.js:1:59005)
    at Fo.beginInvokeDotNetFromJS (blazor.web.js:1:139654)
    at y.invokeDotNetMethodAsync (blazor.web.js:1:4322)
    at S.invokeMethodAsync (blazor.web.js:1:5830)
    at app.js:295:22
    at later (app.js:287:22)

```

**Comment by BrennanConroy** [comment]
Looks similar to https://github.com/dotnet/aspire/issues/5027

**Comment by JamesNK** [comment]
I believe this is fixed in 8.2. Reopen if you see it in 8.2.
